Formula/perl.rb

```diff
@@ -25,26 +26,34 @@ class Perl < Formula
   depends_on "gdbm"
 
   uses_from_macos "expat"
+  uses_from_macos "libxcrypt"
+
+  on_linux do
+    # Configure checks if libnsl is present but does not link to it.
+    # Disabling the libnsl check is not possible without hacky solutions that are prone
+    # to bitrot.  Remove this if upstream provides a more reliable way to skip this check.
+    depends_on "libnsl" => :build
```

Comment:
Just to be clear about this dependency: is it needed when building in CI? Or only for users who need to build `perl` from source (and perhaps happen to have `libnsl` installed)?

Reply:
It's needed if the host system happens to have `libnsl` (either as part of an older `glibc`, or in the standalone package) and the user is building from source.  I know we don't officially support doing that, but since checking for that library is hardcoded this seems like the least painful to make sure this will work on source builds.

Reply:
Hard 👎 for me here -- `perl`'s dep tree it too big to be tacked onto `libnsl` for a problem that is never encountered either in CI or by users who use the default prefix.

Reply:
In that case we really need to find an acceptable way to stop `perl` from searching for `libnsl` then.  Otherwise it will be uninstallable for any user who is not in the default prefix and happens to have `libnsl` somewhere, which could be large fraction of our users.

Probably the safest is just an `inreplace` in `Configure`.  I can push an example so we can look at it.

Reply:
Actually, it turns out in the whole `Configure` script, there, is only one instance of the string " nsl ", which is where it searches for `libnsl`.  So we can get rid of that without having to worry about any other changes to the file.

Reply:
> In that case we really need to find an acceptable way to stop `perl` from searching for `libnsl` then. Otherwise it will be uninstallable for any user who is not in the default prefix and happens to have `libnsl` somewhere, which could be large fraction of our users.

No, we don't need to do anything. That's what "building from source is unsupported" means.

> Probably the safest is just an `inreplace` in `Configure`. I can push an example so we can look at it.

The safest is to not do anything at all about it -- not in the formula, at least. The fact that `Configure` looks for `libnsl` and does nothing with it smells like an upstream bug, and should be reported and fixed there: either it should do something useful with `libnsl` or there should be a `Configure`-time flag that disables it.

Reporting it upstream also ensures that we're not making a mistake about the assessment of the non-use of `libnsl`. If it's not an upstream bug, then we could be introducing bugs of our own by `inreplace`-ing it out.

Reply:
Of course in the endless rounds of testing I failed to notice that simply installing `libnsl` before installing `perl` is sufficient for it to pass the configure step 🤦.  Users building from source will instinctively try to install missing libraries like this so this isn't really a huge hassle and doesn't require them to modify the formula.  Also, `libnsl` is a dependency of Python 3 which is widely used enough that it will likely be installed anyway.

> The safest is to not do anything at all about it -- not in the formula, at least. The fact that Configure looks for libnsl and does nothing with it smells like an upstream bug, and should be reported and fixed there: either it should do something useful with libnsl or there should be a Configure-time flag that disables it.

> Reporting it upstream also ensures that we're not making a mistake about the assessment of the non-use of libnsl. If it's not an upstream bug, then we could be introducing bugs of our own by inreplace-ing it out.

Yeah this is looking more and more like some sort of bug or oversight.  It doesn't make sense to check for a library that isn't linked anywhere.  There is https://metacpan.org/dist/Net-NIS/view/NIS.pod module that probably does link to `libnsl` - but it's not installed in the default modules.  If I had to hazard a guess, `Net::NIS` may have at one point been a default module that was dropped at some point, but the library check was not removed.  

I'm surprised this hasn't already caused issues as their Configure doesn't actually try to link the libraries - it just searches for filenames of the libraries on the disk.  This is a highly dubious strategy to say the least.  We only avoided having an issue on Ubuntu 22.04 because `libnsl2` is included in the default packages that come with the OS.

I am going to open a PR in their GitHub to just delete `nsl` from `libswanted` and see what the response is.  If they confirm it's the right call we can add the fix in a follow up PR.  In the interim the trivial solution for users experiencing build failures is just to install `libnsl` if it's not already installed, so I'm going to drop the `libnsl` fix for now while we find out what upstream says about the whole issue.